### PR TITLE
Add ability to pass a custom message format

### PR DESCRIPTION
**Pull request by jzoldak**
Including job name and parameters for triggering.

This new feature is documented in src/main/resources/com/base2services/jenkins/SqsBuildTrigger/help-manual.jelly

My use case is that we want to be able to pass parameters to a build (e.g. which hash to build, etc.) instead of just poking the job (which looks for changes to determine whether or not to build).

@aaronwalker would you be able to review this PR? Is there anything else I can or should do to get this feature merged into the official plug-in?

### CODE REVIEW

File: src/main/java/com/base2services/jenkins/SqsQueueHandler.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import hudson.util.SequentialExecutionQueue;
 import hudson.util.TimeUnit2;
 
+import java.util.ArrayList;
```
Comment (jzoldak) [inline src/main/java/com/base2services/jenkins/SqsQueueHandler.java]: Reviewer note: 
Changes to this file are unrelated to the new feature. They are improvements to the existing code that makes troubleshooting easier, especially AWS permissions issues.

File: src/test/java/com/base2services/jenkins/GitHubTriggerProcessorTest.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public void shouldNotTriggerWithAFork() throws Exception {
     public void shouldNotTriggerWithBadPayload() throws Exception {
         payload = "{'noRepoInfo': {'empty': 'https://github.com/foo/bar'}}".replace("'", "\"");
         GitHubTriggerProcessor gtp = new GitHubTriggerProcessor();
-        gtp.processGitHubPayload(payload, sbt.getClass());
+        gtp.trigger(payload);
```
Comment (jzoldak) [inline src/test/java/com/base2services/jenkins/GitHubTriggerProcessorTest.java]: Reviewer note:
This worked before because it was relying on the logic in the extractJsonFromPayload function that would have it return null if there was no 'repository' object in the payload. The new code changes will catch this case earlier, in the trigger function.

File: src/main/java/com/base2services/jenkins/github/GitHubTriggerProcessor.java [hunk 3]
```diff
@@ -71,22 +102,117 @@ public void processGitHubPayload(String payload, Class<? extends Trigger> trigge
         }
     }
 
-    private JSONObject extractJsonFromPayload(String payload) {
-        JSONObject repository = null;
+    public JSONObject extractJsonFromPayload(String payload) {
```
Comment (jzoldak) [inline src/main/java/com/base2services/jenkins/github/GitHubTriggerProcessor.java]: Reviewer note:
Changed this function to public so that it can be used in tests.
Also changed its functionality to be more generic - as written it would not extract the payload if there was no "repository" object in it but instead would return null.

**Comment by natefaerber** [comment]
Hi, I'm quite late to the party here.  I'm wondering if there is an example of using the custom payload.  Is this to support SQS coming from somewhere other than GitHub or am I missing the documentation at GitHub where we can customize the payloads sent using the SQS Service for a project?  Thanks.

**Comment by jzoldak** [comment]
Welcome to the party, @natefaerber!

* This is to support SQS coming from somewhere other than GitHub
* More details, and examples of custom payloads can be found in [the help file](../blob/master/src/main/resources/com/base2services/jenkins/SqsBuildTrigger/help-manual.jelly).

**Comment by natefaerber** [comment]
Thanks @jzoldak.  I wasn't sure if I was missing a GitHub custom payload capability.  This is a great feature add to support notifications from non-GitHub projects.
